manifests/server/puppetserver.pp

```diff
@@ -147,6 +150,17 @@
 ) {
   include puppet::server
 
+  # For Puppetserver, certain configuration parameters are version specific.
+  # We need a method to determine what version is installed.
+  if $puppetserver_version {
+    $real_puppetserver_version = $puppetserver_version
+  } elsif versioncmp($facts['puppetversion'], '7.0.0') >= 0 {
+    $real_puppetserver_version = $facts['puppetversion']
```

Comment:
I don't think this is correct. There's puppet-agent 7.27.0 but puppetserver 7.14.0 is the highest version.

I'd simplify it down to just 8.0.0, because that's a safe assumption
```suggestion
  } elsif versioncmp($facts['puppetversion'], '8.0.0') >= 0 {
    $real_puppetserver_version = '8.0.0'
```